pkg/skaffold/build/custom/custom_test.go

```diff
@@ -125,8 +139,57 @@ func TestRetrieveCmd(t *testing.T) {
 	}
 }
 
+func TestGracefulBuildCancel(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("graceful cancel doesn't work on windows")
+	}
+
+	tests := []struct {
+		description string
+		artifact    *latest.Artifact
+		testEnv     []string
+		shouldErr   bool
+	}{
+		{
+			description: "terminate gracefully and exit 0",
+			artifact: &latest.Artifact{
+				ArtifactType: latest.ArtifactType{
+					CustomArtifact: &latest.CustomArtifact{
+						BuildCommand: `bash -c "trap 'echo trap' INT; sleep 2"`,
+					},
+				},
+			},
+		}, {
+			description: "terminate gracefully and kill process",
+			artifact: &latest.Artifact{
+				ArtifactType: latest.ArtifactType{
+					CustomArtifact: &latest.CustomArtifact{
+						BuildCommand: `bash -c "trap 'echo trap' INT; sleep 5"`,
+					},
+				},
+			},
+			shouldErr: true,
+		},
+	}
+
+	for _, test := range tests {
+		testutil.Run(t, test.description, func(t *testutil.T) {
+			builder := NewArtifactBuilder(false, test.testEnv)
+
+			for i := 0; i < 10; i++ {
```

Comment:
please remove 10 iterations. Sorry I meant running it locally to make sure its not flaky.